src/cargo/core/workspace.rs

```diff
@@ -1800,7 +1800,10 @@ impl<'gctx> Packages<'gctx> {
         match self.packages.entry(key.to_path_buf()) {
```

Comment:
Looks like this will be ambiguous when we add workspace support and will be hard to find.  By working around this now, we are laying a trap for ourselves in the future.

Maybe we should have a refactor commit before this one that changes the `key` from being the manifest-dir to being the manifest-path.

Reply:
Yes. Totally argee this is a bit tricky. I will try to refactor it and change the key.